### ISSUE TITLE
Grouped Window List Applet: window preview partially occluded by panel

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [linuxmint/cinnamon#8578]
```
 * Cinnamon version (cinnamon --version) 4.0.10
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Mint 19.1
 * Graphics hardware *and* driver used: Intel UHD Graphics 620 Mesa 3
 * 32 or 64 bit: 64
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
 ```

**Issue**

Sometimes I see the following.
![image](https://user-images.githubusercontent.com/47056543/58061073-728c6a00-7b6c-11e9-8a1c-1b569af83e01.png)
I am unsure whether the problem besets only some applications. Indeed in the case of at least two applications, namely, SublimeText and Firefox, the problem is intermittent. For some times thinks work as they should, thusly:
![image](https://user-images.githubusercontent.com/47056543/58061348-b469e000-7b6d-11e9-8ee4-cc469888dd77.png)

EDIT: another (intermittently?) affected application is Mint's System Settings.
**Steps to reproduce**

![image](https://user-images.githubusercontent.com/47056543/58061124-b1222480-7b6c-11e9-97e0-86b68dfe55bd.png)

Desktop Theme: Mint-Y-Dark

EDIT: Further relevant settings are as follows.

![image](https://user-images.githubusercontent.com/47056543/58063461-4bd33100-7b76-11e9-81d9-59aa719df5e1.png)

**Expected behaviour**

The thumbnail should never underlie the panel.

**Other information**

* My first screenshot does indeed show a panel - not a dock. I have made the panel resemble a dock.
* ~~When I tried out the alpha of the next version of Mint, the problem seemed to have gone. I thought that it might nevertheless be of use for this bug report to exist.~~

**Comment by basilky** [comment]
This also affects other applets. The following is a screenshot of 'Download and upload speed' applet. I have panel setting 'Intelligently hide panel' enabled.
![Screenshot from 2019-05-31 22-19-40](https://user-images.githubusercontent.com/22705230/58722780-f6a8e200-83f5-11e9-84b0-721fae07a94c.png)

**Comment by jaszhix** [comment]
Try disabling the transparent panels extension.

**Comment by basilky** [comment]
Disabled transparent panels. Still no luck.

**Comment by LinuxOnTheDesktop** [comment]
Also (basilky wrote: 'Disabled transparent panels. Still no luck') this inconsistent behaviour - whereby sometimes the window preview is wholly above the panel and sometimes it is not - surely is not the intended behaviour for transparent panels. So, this _is_ a Cinnamon bug and not, it seems, a bug in any particular theme.

**Comment by LinuxOnTheDesktop** [comment]
Given

&nbsp;&nbsp;&nbsp;(1) this problem with panels

and

&nbsp;&nbsp;&nbsp;(2) that [clicks can go accidentally through icons and tooltips malfunction](https://github.com/linuxmint/nemo/issues/2060)

and

&nbsp;&nbsp;&nbsp;(3) the fact that window previews are 'jittery' in that they disappear too easily[1]

using Mint Cinnamon isn't very pleasant at present (and that's not to mention various problem with the MintMenu, which have also been filed on the bug tracker).

&nbsp;&nbsp;&nbsp;Also, has the video recorder, which is useful for reporting bugs, been fixed yet, please?

[1] Perhaps I should file a separate bug report against the grouped-window-list applet.

**Comment by LinuxOnTheDesktop** [comment]
The problem persists on the alpha, I'm afraid.

```
$  cinnamon --version
Cinnamon 4.2.0
```

**Comment by basilky** [comment]
There is no problem when the session is started and the issue comes up only after using the system for 10 to 15 minutes. Also, a right click on the extension makes it look normal for sometime. But, after some time panel again obstructs the extension.

**Comment by LinuxOnTheDesktop** [comment]
It would be nice were this problem to be fixed in Mint 19.2.

Does the comment above by basilky provide anything to go on in trying to solve the problem?

**Comment by LinuxOnTheDesktop** [comment]
[xsession.txt](https://github.com/linuxmint/cinnamon/files/3409354/xsession.txt)

**Comment by LinuxOnTheDesktop** [comment]
![image](https://user-images.githubusercontent.com/47056543/61596439-a121d200-abfb-11e9-874c-088ffe11d993.png)


```
$  cinnamon --version
Cinnamon 4.2.2
```

**Comment by basilky** [comment]
@LinuxOnTheDesktop, can you please report this issue on https://github.com/linuxmint/mint19.2-beta/issues? 
Mint 19.2 is shipping with cinnamon 4.2 and this issue might be present in the beta also.

**Comment by LinuxOnTheDesktop** [comment]
@basilky 
I propose that you do it. For, I think it would be better coming from someone other than me. You can link to the report on this page.

**Comment by basilky** [comment]
Following is a screen record of the issue. 
https://youtu.be/JizvJan2syo

**Comment by LinuxOnTheDesktop** [comment]
I'd be happy to further investigate this bug myself, but I'd need some pointers.

**Comment by mtwebster** [comment]
I've been trying to reproduce this without luck so far, though it sounds like a common theme here is auto- or intelligently-hiding panels.

The best help I could have here is a reliable way to reproduce it.

**Comment by LinuxOnTheDesktop** [comment]
To the end of helping reproduce the problem, I supply the following.

Here is _my Cinnamon theme_: [link](https://app.box.com/s/syew178btehir02tn8r09l8x79x477cc)

Here are _images of some relevant settings_

![image](https://user-images.githubusercontent.com/47056543/69671791-a5209d80-108e-11ea-98f0-a6e929ba74de.png)

![image](https://user-images.githubusercontent.com/47056543/69671848-bf5a7b80-108e-11ea-8ff6-3dac37daeb9b.png)

![image](https://user-images.githubusercontent.com/47056543/69671916-dbf6b380-108e-11ea-949b-f471619d5294.png)

![image](https://user-images.githubusercontent.com/47056543/69671982-ffb9f980-108e-11ea-8bbc-1e59b6230a28.png)


Here is how the relevant _panel_ of mine (when not hidden) looks:

![image](https://user-images.githubusercontent.com/47056543/69671456-fa0fe400-108d-11ea-9f08-e506f729724c.png)

**Comment by LinuxOnTheDesktop** [comment]
I had wondered whether the problem only occurs when I have Hi-DPI scaling enabled.

(When I have that scaling enabled my screen resolution is in effect (2560 / 2) x (1440 / 2) = 1280 x 720. However, without DPI x 2 scaling, my computer becomes unusuable with Cinnamon - because, for one thing, window titlebars are much too small.)

I find the answer is: no. Even with Hi-DPI scaling disabled (and having restarted Cinnamon, but, admittedly, not rebooted), the problem persists (intermittently).

**Comment by LinuxOnTheDesktop** [comment]
This problem seems fixed in Mint 19.3. Hooray!
